### PR TITLE
runitor: init at 0.9.2

### PR DESCRIPTION
**Pull request by bdd**
###### Description of changes
New package: runitor at v0.9.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/runitor/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-LuCxn4j0MlnJjSh3d18YNzNrtbqtMPxgkZttqKUGJd4";
+  };
+
+  ldflags = [
+    "-s" "-w" "-X main.Version=v${version}"
+  ];
+
+  doInstallCheck = true;
+  installCheckPhase = ''
```
Comment (stehessel) [inline pkgs/tools/system/runitor/default.nix:20]: You can use `passthru.tests.version` to test the version. See https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/cluster/openshift/default.nix#L55 for an example.

**Review by stehessel — CHANGES_REQUESTED** [review]
Also it looks like the passthru tests are currently failing.

File: pkgs/tools/system/runitor/default.nix (hunk 1 above)
Comment (bdd) [reply pkgs/tools/system/runitor/default.nix]: [884e153](https://github.com/NixOS/nixpkgs/pull/170566/commits/884e153c0ab5be8913d92084b2fbbb94fcab7ced)

**Comment by bdd** [comment]
> Also it looks like the passthru tests are currently failing.

`go test ./...` invocation runs unit tests that require loopback networking to bring up stub listeners to test the client behavior. IIUC, networking is entirely disabled in OfBorg runners, hence those unit tests fail.

File: pkgs/tools/system/runitor/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ lib, buildGoModule, fetchFromGitHub, testers, runitor }:
```
Comment (IvarWithoutBones) [inline pkgs/tools/system/runitor/default.nix:1]: You shouldn't need to use `testers` here as `testVersion` is already exposed, I think:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, testVersion, runitor }:
```

Comment (bdd) [reply pkgs/tools/system/runitor/default.nix:1]: I think the preferred way is to use `testers.testVersion` now.

There was a tree-wide search-and-replace recently: f1c7f19e49190cf58041908f9c194431e2513688

...and these are the current stats:
```
nixpkgs% rg -l '= testers.testVersion' pkgs | wc -l
62
nixpkgs% rg -l '= testVersion' pkgs | wc -l
5
```

**Comment by cole-h** [comment]
I don't know why, yet, but the tests experience a network failure on the ofborg builders... I don't have much time to look into why that is this weekend, so... For the moment, would you mind disabling the tests (or even just those specific tests, if possible...)? Feel free to add a `TODO(cole-h): figure out why tests fail loopback networking in ofborg` comment above it, if you so desire. (FWIW: It builds just fine on my local machine, no networking issues whatsoever.)

**Comment by bdd** [comment]
> I don't know why, yet, but the tests experience a network failure on the ofborg builders... I don't have much time to look into why that is this weekend, so... For the moment, would you mind disabling the tests (or even just those specific tests, if possible...)? Feel free to add a `TODO(cole-h): figure out why tests fail loopback networking in ofborg` comment above it, if you so desire. (FWIW: It builds just fine on my local machine, no networking issues whatsoever.)

Alright. I was gonna skip these tests but then I hit a bug in `go-modules/generic/default.nix` w.r.t how `$checkFlags` is passed to `buildGoDir` helper, which doesn't expect more than 2 parameters but a checkFlags list may get expanded to a space separated string, which will get interpreted as multiple arguments. Will send a PR for that as well.

```
diff --git a/pkgs/development/go-modules/generic/default.nix b/pkgs/development/go-modules/generic/default.nix
index 502a981596e..dd829e8dbd6 100644
--- a/pkgs/development/go-modules/generic/default.nix
+++ b/pkgs/development/go-modules/generic/default.nix
@@ -244,7 +244,7 @@ let
       runHook preCheck
░
       for pkg in $(getGoDirs test); do
-        buildGoDir test $checkFlags "$pkg"
+        buildGoDir "test $checkFlags" "$pkg"
       done
░
       runHook postCheck

```

**Pull request merged**